browser_use/browser/session.py

```diff
@@ -2667,24 +2743,50 @@ async def _get_updated_state(self, focus_element: int = -1) -> BrowserStateSumma
 			# 	)
 
 			try:
-				screenshot_b64 = await self.take_screenshot()
+				self.logger.debug('📸 Starting screenshot...')
+				# Reasonable timeout for screenshot
+				screenshot_b64 = await asyncio.wait_for(
+					self.take_screenshot(),
+					timeout=15.0,  # 15 second reasonable timeout
```

Comment:
this is not needed because screenshot is already wrapped in `@retry`

Reply:
True